### PR TITLE
Update to 2.10.4

### PR DESCRIPTION
**Pull request by JeanChristopheMorinPerso**
Diff between 2.10.1 and 2.10.4: https://github.com/pybind/pybind11/compare/v2.10.1..v2.10.4.

Required by https://github.com/AnacondaRecipes/dpctl-feedstock/pull/3.

### CODE REVIEW

**Review by cbouss — COMMENTED** [review]
LGTM, with abs.yaml removed

File: abs.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# the conda-build parameters to use for disabling --skip-existing
+build_parameters:
```
[the remark this answers is not in the document]
Comment (JeanChristopheMorinPerso) [reply abs.yaml]: I just pushed a commit to remove this file and we get an error like `failed with error: [Errno 2] No such file or directory: '/upload/noarch/pybind11-abi-4-hd3eb1b0_0.tar.bz2'`. Is this a conda-build or a CI bug or something like that?

**Comment by JeanChristopheMorinPerso** [comment]
I added the `abs.yaml` file back again to fix issues where `conda-build` would try to copy(?) the skipped `pybind11-abi-4`. See https://github.com/conda/conda-build/issues/4856 and https://cloud.prefect.io/anaconda/flow-run/22c8413c-7927-4673-ba48-0adddff4e2f8?logs for details.

**Review by cbouss — APPROVED** [review]

**Review by ELundby45 — APPROVED** [review]

File: abs.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (skupr-anaconda) [reply abs.yaml:3]: Should we make a build number for pybind11-abi-4 separately? Because it's a new build for it even if we do not update it really

**Pull request merged**